core/backend/src/IModelDb.ts

```diff
@@ -1739,11 +1739,81 @@ export namespace IModelDb { // eslint-disable-line no-redeclare
      * @see getElementJson
      */
     private tryGetElementJson<T extends ElementProps>(loadProps: ElementLoadProps): T | undefined {
-      try {
-        return this._iModel.nativeDb.getElement(loadProps) as T;
-      } catch (err: any) {
+      const { id, code, federationGuid } = loadProps;
+
+      if (!this._iModel.isOpen)
+        return undefined;
+
+      let elementId: string | undefined;
+
+      if (id !== undefined) {
+        elementId = id;
+      } else if (federationGuid !== undefined) {
+        elementId = this._iModel.withPreparedSqliteStatement("SELECT Id FROM bis_Element WHERE FederationGuid=?", (stmt: SqliteStatement) => {
+          stmt.bindGuid(1, federationGuid);
+          return stmt.nextRow() ? stmt.getValueId(0) : undefined;
+        });
+      } else if (code !== undefined) {
+        elementId = this._iModel.withPreparedSqliteStatement("SELECT Id FROM bis_Element WHERE CodeSpecId=? AND CodeScopeId=? AND CodeValue=? LIMIT 1", (stmt: SqliteStatement) => {
+          stmt.bindId(1, code.spec);
+          stmt.bindId(2, code.scope);
+          code.value !== undefined ? stmt.bindString(3, code.value) : stmt.bindNull(3);
+          return stmt.nextRow() ? stmt.getValueId(0) : undefined;
+        });
+      }
+
+      if (elementId === undefined)
+        return undefined;
+
+      const elementProps = this._iModel.withPreparedStatement("SELECT $ FROM Bis.Element WHERE ECInstanceId=? OPTIONS USE_JS_PROP_NAMES DO_NOT_TRUNCATE_BLOB", (statement: ECSqlStatement) => {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        statement.bindId(1, elementId!); // elementId can't be null here, but eslint recognizes it as an error.
+
+        if (statement.step() !== DbResult.BE_SQLITE_ROW)
+          return undefined;
+
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+        return mapNativeElementProps(JSON.parse(statement.getValue(0).getString()), loadProps) as T;
+      });
+
+      if (!elementProps)
         return undefined;
+
+      if (loadProps.wantGeometry || loadProps.wantBRepData) {
```

Comment:
This should just be if (loadProps.wantGeometry)...wantBRepData is only applicable when wantGeometry is true.